### PR TITLE
feat(query): Support `show dictionaries` DQL. 

### PR DESCRIPTION
**Pull request by Winnie-Hong0927**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

We continue to improve the logical plan and binder of dictionary.
This allows us to execute Show-Dictionaries DQL statements in MySql.
For example:
1. Show all dictionaries in MySql: `show dictionaries;`
2. Show dictionaries in specific database: `show dictionaries from | in db_name;`
3. Show dictionaries with conditions: `show dictionaries like 'd%'`

fix: #16551

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16602)

### CODE REVIEW

File: src/query/sql/src/planner/binder/ddl/dictionary.rs [hunk 1]
```diff
@@ -383,4 +389,55 @@ impl Binder {
...
+            .with_column("attribute_types AS Attribute_Types")
+            .with_column("source AS Source")
+            .with_column("comment AS Comment");
+
+        select_builder
+            .with_order_by("database")
+            .with_order_by("name");
+
+        if database.len() != 0 {
+            select_builder.with_filter(format!("database = '{database}'"));
```
Comment (b41sh) [inline src/query/sql/src/planner/binder/ddl/dictionary.rs:423]: we can write like this
```rust
if let Some(db) = database {
    select_builder.with_filter(format!("database = '{}'", db.name));
}
```

File: src/query/storages/system/src/dictionaries_table.rs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            ColumnBuilder::with_capacity(&DataType::Array(Box::new(DataType::String)), 0);
+
+        let mut sources = vec![];
+        let mut comments = vec![];
+        let mut created_ons = vec![];
+        let mut updated_ons = vec![];
+
+        let mgr = CatalogManager::instance();
+        let catalogs = mgr.list_catalogs(&tenant, ctx.session_state()).await?;
+        for catalog in catalogs {
```
Comment (b41sh) [inline src/query/storages/system/src/dictionaries_table.rs:84]: don't need check all catalogs, use default catalog instead. `self.ctx.get_default_catalog()`

File: src/query/storages/system/src/dictionaries_table.rs [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                    db_names.push(database.get_db_name().to_string());
+
+                    names.push(dict_name.clone());
+
+                    let dict_name_ident = DictionaryNameIdent::new(
+                        tenant.clone(),
+                        DictionaryIdentity::new(db_id, dict_name.clone()),
+                    );
+                    let reply = catalog.get_dictionary(dict_name_ident).await?.unwrap();
+                    let dict_id = reply.dictionary_id;
```
Comment (b41sh) [inline src/query/storages/system/src/dictionaries_table.rs:103]: we can ignore `dict_id`

File: src/query/sql/src/planner/binder/ddl/dictionary.rs [hunk 4]
```diff
@@ -383,4 +389,53 @@ impl Binder {
...
+            .with_column("attribute_names AS Attribute_Names")
+            .with_column("attribute_types AS Attribute_Types")
+            .with_column("source AS Source")
+            .with_column("comment AS Comment");
+
+        select_builder
+            .with_order_by("database")
+            .with_order_by("name");
+
+        let mut db_name = String::new();
```
Comment (b41sh) [inline src/query/sql/src/planner/binder/ddl/dictionary.rs:417]: use `self.check_database_exist` function to get db name. 
```rust
let database = self.check_database_exist(&None, database).await?;`
```

**Review by b41sh — APPROVED** [review]
LGTM, thanks for your contribution. @Winnie-Hong0927

**Comment by BohuTANG** [comment]
@soyeric128 New feature is up, please help with the documentation, thanks!

**Comment by BohuTANG** [comment]
@Winnie-Hong0927 Thanks for your contribution!

**Pull request merged**